Editorial (#146)
danielfett/draft-dpop: main.md

   pseudorandom data or by using a version 4 UUID string according to [@RFC4122].
   The `jti` can be used by the server for replay
   detection and prevention, see (#Token_Replay).
 * `htm`: The HTTP method of the request to which the JWT is
   attached, as defined in [@!RFC7231].
 * `htu`: The HTTP request URI ([@RFC7230, section 5.5]), without query and
   fragment parts.
 * `iat`: Creation timestamp of the JWT ([@RFC7519, section 4.1.6]).

When the DPoP proof is used in conjunction with the presentation of an access token, see 
(#protected-resource-access), the DPoP proof MUST also contain the following claim:

* `ath`: hash of the access token.
   The value MUST be the result of a base64url encoding (with no padding) the SHA-256

>    hash of the ASCII encoding of the associated access token's value.

Reviewer: yes, just the access token

>    hash of the ASCII encoding of the associated access token's value.

Reviewer: base64url can/should come from RFC7515 https://datatracker.ietf.org/doc/html/rfc7515#section-2 which is admittedly a kinda strange place for it but more precise for this usage

## Checking DPoP Proofs {#checking}

To validate a DPoP proof, the receiving server MUST ensure that

 1. that there is not more than one `DPoP` HTTP request header field,
 1. the header field value is a well-formed JWT,
 1. all required claims per (#DPoP-Proof-Syntax) are contained in the JWT,
 1. the `typ` JOSE header parameter has the value `dpop+jwt`,
 1. the `alg` JOSE header parameter indicates an asymmetric digital
    signature algorithm, is not `none`, is supported by the
    application, and is deemed secure,
 1. the JWT signature verifies with the public key contained in the `jwk`
    JOSE header parameter,
 1. the `jwk` JOSE header parameter does not contain a private key,
 1. the `htm` claim matches the HTTP method of the current request,

>  1. the `htu` claim matches the HTTPS URI value for the HTTP
>     request in which the JWT was received, ignoring any query and
>     fragment parts,

Reviewer 2: ```suggestion
 1. the `htu` claim matches the https URI of the HTTP
    request in which the JWT was received, ignoring any query and
    fragment parts,
```

iirc the protocol is `HTTP`. The URI scheme is `https`. 

Probably a more precise statement would be

> matches the target URI [@RFC7231, section 2] of the current request
> ignoring any query and fragment parts, and with the https URI scheme

Consider that the latest HTTP semantics disentangles the request URI from 
HTTP messaging (see https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#rfc.section.7.1.p.2)

Protected resources simultaneously supporting both the `DPoP` and `Bearer` 
schemes need to update how evaluation of bearer tokens is performed to prevent 

> downgraded usage of a DPoP-bound access tokens.

Reviewer 3: access tokens -> access token.

Author: and done f51a9d998120a95c0f6e8fa64195f566293ab7e2

# Authorization Server-Provided Nonce {#ASNonce}

This section specifies a mechanism using opaque nonces provided by the server
that can be used to limit the lifetime of DPoP proofs.
Without employing such a mechanism, a malicious party controlling the client

> (including potentially the end user)

Reviewer 3: end user -> end-user
